python/ray/data/_internal/execution/resource_manager.py

```diff
@@ -649,3 +653,18 @@ def update_usages(self):
             # We don't limit GPU resources, as not all operators
             # use GPU resources.
             self._op_budgets[op].gpu = float("inf")
+
+        from ray.data._internal.execution.operators.base_physical_operator import (
+            AllToAllOperator,
+        )
+
+        # An all-to-all operator waits for all its input operator’s outputs before
+        # processing data. This often forces the input operator to exceed its object
+        # store memory budget. To prevent deadlock, we disable object store memory
+        # backpressure for the input operator.
```

Comment:
This won't be the case for long 

https://anyscale1.atlassian.net/browse/DATA-571

Reply:
We still need all of the inputs to produce an output, though, right?